Add Maven support, add server modified block support, fix compilation issues

TheGreatRambler opened the pull request.

Jumper251 commented:
This pull request is difficult to review because it contains a lot of changes for different features/issues.
I am not sure which files are just reformatted and which contain different code.
I also noticed you replaced all packetwrappers and some of my changes to them in the process.

TheGreatRambler commented:
It's possible this is uncommitable. A significant amount of the code was changed, and that's my bad. Firstly, I added Maven because tbh I didn't know how to build the codebase otherwise. Next, I had to modify the PacketWrappers because the ones you included didn't have the BlockChange wrapper. Then, the code I modified used a different formatter than yours, and because you didn't provide your formatter I just made every file consistent. I would ask that you provide the clang-format settings you used in your codebase so I can fix that. Otherwise, the other additions are basically required for this PR and I don't see any problem in introducing them, as they make the codebase much more maintainable.

TheGreatRambler commented:
Actually, I'm going to maintain this fork myself and separate it from this project. I'm planning on making some significant changes.
